joomla/joomla-cms
# Make media field initialize itself when a new subform row is added

Pull Request for Issue # .

### Summary of Changes

Subform repeatable has a function `fixScripts` which calls a few js initialization functions on a few of the field types that require it. There is a now a better way to handle that kind of thing by having any field that may need initialization listen for an event fired by subform repeatable. In this PR, I am transferring responsibility of one such initialization from subform repeatable to the correct field type itself. 

Note: this PR doesn't really fix anything. the media field type already works with repeatable subforms. This just does it in a somewhat better way. 

### Testing Instructions

Set up a subform containing a media field. There's a subform in the redirect plugin that's simple to use for this purpose. 

Open up the form and create new rows. The media field in each new row should function properly. 

### Expected result

Media field functions normally.

### Actual result

Media field functions normally.

### Documentation Changes Required

None
